### ISSUE TITLE
working in Notebook / CLI but not in jupyterlab?

### ISSUE THREAD

**Issue by kinoc** [Jaykul/Jupyter-PowerShell#17]
While using the C# core for another project I came across the problem of 
the kernel working great in Notebook but not Jupyterlab. After much digging I found the issue is the same as the following link:
[https://github.com/jupyterlab/jupyterlab/issues/470](https://github.com/jupyterlab/jupyterlab/issues/470)
Basically you need to modify the KernelInfoHandler to send PublishStatus(message, ioPub, KernelState.Busy) before the message and PublishStatus(message, ioPub, KernelState.Idle) after it is done in HandleMessage just like you do for an ExecuteRequestHandler. You have to do this for all messages you choose to handle. The new system going forward does not consider the kernel to be active until it gets the KernelState.Idle on the ioPub channel for every message it sends forward and gets handled. 

Just thought someone might like to know.

**Comment by Jaykul** [comment]
yeah, in particular the lab *console* is broken, I need to add support for a `history `message that I somehow missed in the Jupyter docs.
